shared/studio/components/headerNav/elements.tsx

```diff
@@ -109,8 +113,27 @@ export function HeaderNavCol<LinkProps>({
   showCursor = false,
   closeDropdown,
   itemGroups,
+  currentOrgSlug,
+  currentInstanceName,
+  currentDatabaseName,
+  selectedItems,
   action,
 }: HeaderNavColProps<LinkProps>) {
+  const getIsSelected = (type: string, label: string) => {
+    if (type === "Personal Org" || type === "Orgs") {
+      return label === currentOrgSlug;
+    } else if (type === "Instances") {
+      return (
+        currentOrgSlug === selectedItems![0] && currentInstanceName === label
+      );
+    } else
+      return (
+        currentOrgSlug === selectedItems![0] &&
+        currentInstanceName === selectedItems![1] &&
+        currentDatabaseName === label
+      );
+  };
+
```

Comment:
I think this logic needs to be outside this component, since it's also used in the local ui which doesn't have the same org/instance/db url structure, so this breaks. Probably the easiest way would be to add a `current`/`checked` boolean to the items like the `selected` boolean.